options/flags.go

```diff
@@ -59,6 +59,12 @@ const (
 
 	// FlagFormat is the flag name for specifying output format.
 	FlagFormat = "format"
+
+	// FlagBase is the flag name for specifying a dependency-diff base.
+	FlagBase = "base"
```

Comment:
I'm curious if base and head make sense to a user. Is this a common naming convention if bitbucket and gitlab as well?

Is there better term for these otherwise (`start-commit`, `end-commit`, etc)

@olivekl may have some advice

Reply:
> I'm curious if base and head make sense to a user. Is this a common naming convention in bitbucket and gitlab as well?
> 
> Are there better terms for these otherwise (`start-commit`, `end-commit`, etc)
> 
> @olivekl may have some advice

Thanks. I've considered using some flag like `--ref` to receive the base and head. It is clear for receiving a branch name reference, but is `commitSHA` also a `ref`? I'm not sure if `--ref` will also lead to ambiguity.

Or we just create two flags `--ref` and `--sha` so users can use either of them.

Reply:
sync from the bi-weekly meeting:

`--base` and `--head` don't have the ambiguity of whether using a `reference` or using a `commitSHA`, and might be good enough for users to correctly use them if there is a clear document. 

Here's my doc in the `--help` option:
`--base: the base branch name reference or the base commitSHA to check. valid input examples: 
			"main" (branch name), "SHA_VALUE" (commitSHA)`,
			
`--head: the head branch name reference or the head commitSHA to check. valid input examples: 
			"dev" (branch name), "SHA_VALUE" (commitSHA)`,
			
Using "SHA_VALUE" to represent a hash string since it's dangerous to expose a real SHA in the doc.

Reply:
is `base` and `head` a GitHub concept or is it universally used across other platforms like GitLab, BitBuckets, etc?

Reply:
I searched `base` and `head` for `GitLab` and `BitBuckets` online,

https://gitlab.com/gitlab-org/gitlab/-/issues/209071
https://stackoverflow.com/questions/53889502/base-vs-default-branch-in-bitbucket-version-of-git
https://stackoverflow.com/questions/2304087/what-is-head-in-git

found some issues and stackoverflow answers where the developers use BASE and HEAD to refer to code commits and branches, but i didn't find an official document using these two phrases.

Reply:
Are the terms being used as they normally would be in Git? If so, I think it's safe to assume they're well known enough and not platform dependent.

Reply:
> Are the terms being used as they normally would be in Git? If so, I think it's safe to assume they're well known enough and not platform dependent.

Yes, `base` and `head` are normally used in Git, and the [GitHub Dependency Review doc](https://docs.github.com/en/rest/dependency-graph/dependency-review#get-a-diff-of-the-dependencies-between-commits) also makes these two explicit.

Reply:
resolving this conversation